src/units/drawer.cpp

```diff
@@ -62,6 +62,31 @@ std::unique_ptr<image::locator> get_orb_image(orb_status os)
 	return std::make_unique<image::locator>(game_config::images::orb + "~RC(magenta>" + color + ")");
 }
 
+/**
+ * Assemble a two-color orb for an ally's unit, during that ally's turn.
+ * Returns nullptr if the preferences have been configured to hide allied orbs.
+ * Returns a one-color orb if the preferences are to show allied orbs, but not this status.
+ */
+std::unique_ptr<image::locator> get_playing_ally_orb_image(orb_status os)
+{
+	if(!orb_status_helper::prefs_show_orb(orb_status::allied))
+		return nullptr;
+
+	if(!preferences::show_status_on_ally_orb())
+		return get_orb_image(orb_status::allied);
+
+	if(os == orb_status::disengaged)
+		os = orb_status::partial;
```

Comment:
Going to rework this conditional, which might push this PR back until after #7261. The conditional should be replaceable with a comment, because `orb_status_helper::get_orb_color` should have logic to provide a single color for the disengaged state. As noticed in #7261, there's currently an option in game_config.cfg that's a leftover from the idea of using a new single-color orb for disengaged.

Backporting this PR is in any case postponed until after 1.16.7, because of the new translatable string in the preferences dialog.

Reply:
Outdated, although it's not be automatically marked as such because this line didn't change. It's now covering the edge-case that the user has disabled the normal partial orb, but still wants to show the disengage orb as a reminder for the few units that have that skill.